### ISSUE TITLE
Fix some broken ref links in docs

### ISSUE THREAD

**Issue by sjspielman** [AlexsLemonade/scpca-docs#122]
While working on adding ADT content, I found some bad cross references. These should be fixed!

Here's some relevant output from locally rendering the `sphinx` docs as a general guide for places to check:

```
download_files.md:6: WARNING: 'myst' cross-reference target not found: 'multiplexed-sample-libraries' [myst.xref_missing]
download_files.md:34: WARNING: 'myst' cross-reference target not found: 'spatial-transcriptomics-libraries' [myst.xref_missing]
download_files.md:95: WARNING: 'myst' cross-reference target not found: 'download-folder-structure-for-project-downloads' [myst.xref_missing]
download_files.md:105: WARNING: 'myst' cross-reference target not found: 'metadata' [myst.xref_missing]

getting_started.md:134: WARNING: 'myst' cross-reference target not found: 'special-considerations-for-cite-seq-experiments' [myst.xref_missing]
getting_started.md:289: WARNING: 'myst' cross-reference target not found: 'quality-control' [myst.xref_missing]
etting_started.md:289: WARNING: 'myst' cross-reference target not found: 'normalization' [myst.xref_missing]
getting_started.md:289: WARNING: 'myst' cross-reference target not found: 'dimensionality-reduction' [myst.xref_missing]

processing_information.md:7: WARNING: 'myst' cross-reference target not found: 'selective-alignment' [myst.xref_missing]
processing_information.md:7: WARNING: 'myst' cross-reference target not found: 'reference-transcriptome-index' [myst.xref_missing]
processing_information.md:76: WARNING: 'myst' cross-reference target not found: 'selective-alignment' [myst.xref_missing]
/processing_information.md:76: WARNING: 'myst' cross-reference target not found: 'alevin-fry-parameters' [myst.xref_missing]
processing_information.md:81: WARNING: 'myst' cross-reference target not found: 'filtering-cells' [myst.xref_missing]
processing_information.md:104: WARNING: 'myst' cross-reference target not found: 'selective-alignment' [myst.xref_missing]
processing_information.md:104: WARNING: 'myst' cross-reference target not found: 'alevin-fry-parameters' [myst.xref_missing]
processing_information.md:108: WARNING: 'myst' cross-reference target not found: 'combining-adt-counts-with-rna-counts' [myst.xref_missing]
processing_information.md:108: WARNING: 'myst' cross-reference target not found: 'filtering-cells' [myst.xref_missing]
processing_information.md:135: WARNING: 'myst' cross-reference target not found: 'reference-transcriptome-index' [myst.xref_missing]

sce_file_contents.md:145: WARNING: 'myst' cross-reference target not found: 'cell-metrics' [myst.xref_missing]
sce_file_contents.md:180: WARNING: 'myst' cross-reference target not found: 'experiment-metadata' [myst.xref_missing]
sce_file_contents.md:200: WARNING: 'myst' cross-reference target not found: 'cell-metrics' [myst.xref_missing]
```

**Comment by sjspielman** [comment]
Great news, virtually all of what I caught above was actually be an artifact of my local `sphinx` being sad about `#same-document-xref` as I ran it in with nitpicky on! After checking these manually, only one was funky and should be fixed - 

```
sce_file_contents.md:145: WARNING: 'myst' cross-reference target not found: 'cell-metrics' [myst.xref_missing]
```

I went through the docs manually and didn't find anything else (which doesn't mean there isn't something else...).

**Comment by sjspielman** [comment]
closed by #123
